### DISCUSSION TITLE
DB keeps recreating Classification Images

### DISCUSSION THREAD

**Discussion by Apocangel** [d8ahazard/sd_dreambooth_extension#899] [Q&A]
tl;dr I have a set of classification images already generated by DB, but every time I try to train, it starts creating new ones. 
Dataset is unchanged. Settings are the same.
**Any way to force DB to use existing ones?**

The long story:

I'm on latest on all.
running on Auto1111 SD

I am training a LoRA model with a dataset of about 1000 images, and need to create about 5 classification images. Thats 5k Classification images to generate. I pre-generated them all which took about 40 hours with my hardware. Then I tried starting the training, but it started generating new classification images all over again in the same folder it created them previously (./classifiers_0/). I let it run and after another 40 hours, it started training. I had to stop to update SD. After restart, it started generating new classification images a third time.

I tried removing some images from dataset, and it still tries to regenerate new ones, albeit a lower number. I tried removing some of dataset to force it to regenerate only the missing ones, but it tries to regenerate the whole set.

Any way to force DB to use existing ones?

**Comment by d8ahazard** [comment]
If pointing at a directory of existing images, the app will just use that
directory. But, class images now need to match the resolution of their base
input images. Without any other information, my best guess is you maybe
generated class images before bucketing was implemented, and then I fixed
bucketing, which makes the requirement change.

**Reply by Apocangel** [reply]
It is possible that the change in bucketing cause this. The resolution of both is the same. If there is any other information I can provide that would help, please let me know. It is currently generating new images, so I will post an update in about a day.

Thnx for the reply, btw.

**Reply by moraaaaaan** [reply]
have you solve this issue? i'm having the same problem right now, keeps generating images no matter what.

Thanks @d8ahazard @ArrowM

**Comment by Apocangel** [comment]
Nope. Nothing has fixed the problem so far.

**Comment by ArrowM** [comment]
It's basically what D8 said. The class images are linked to the instance images. There have been some bugs related to how to it found them in the past. However, on the latest main commit, class images should be working properly.

If you use class images you need to set the # of class images per instance for each concept in the UI. for example if you set it to 1, it will try to find 1 class image that matches each instance pic. The resolutions must match the post-bucket Ed instance pictures. If you are using captions, the captions must match too.

The extension will create the class images that it needs and can't find.
